### PR TITLE
6898 get iLeapp to work on other disk images and logical file sets

### PR DESCRIPTION
**Pull request by markmckinnon**

### CODE REVIEW

File: Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappAnalyzerIngestModule.java [hunk 1]
```diff
@@ -112,65 +120,134 @@ public void startUp(IngestJobContext context) throws IngestModuleException {
...
-            }
+        ProcessResult fileProcessorResult = iLeappFileProcessor.processFiles(dataSource, moduleOutputPath, iLeappFile);
+
+        if (fileProcessorResult == ProcessResult.ERROR) {
+            return;
+        }
+    }
 
-            ProcessResult fileProcessorResult = iLeappFileProcessor.processFiles(dataSource, moduleOutputPath, iLeappFile);
+    private void processILeappFs(Content dataSource, Case currentCase, DataSourceIngestModuleProgress statusHelper, String directoryToProcess) {
```
Comment (kellykelly3) [inline Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappAnalyzerIngestModule.java:214]: Missing method header

File: Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappAnalyzerIngestModule.java [hunk 2]
```diff
@@ -221,10 +298,21 @@ private ProcessBuilder buildiLeappCommand(Path moduleOutputPath, String sourceFi
         return processBuilder;
     }
 
+    private ProcessBuilder buildiLeappListCommand(Path moduleOutputPath) {
```
Comment (kellykelly3) [inline Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappAnalyzerIngestModule.java:301]: Method header.

File: Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappAnalyzerIngestModule.java [hunk 3]
```diff
@@ -264,4 +357,109 @@ private void addILeappReportToReports(Path iLeappOutputDir, Case currentCase) {
 
     }
 
+    /*
+     * Reads the iLeapp paths file to get the paths that we want to extract
+     *
+     */
+    private List<String> loadIleappPathFile(Path moduleOutputPath) throws FileNotFoundException, IOException {
```
Comment (kellykelly3) [inline Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappAnalyzerIngestModule.java:364]: Add @param to method header.

File: Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappAnalyzerIngestModule.java [hunk 4]
```diff
@@ -264,4 +357,109 @@ private void addILeappReportToReports(Path iLeappOutputDir, Case currentCase) {
...
+                }
+                iLeappPathsToProcess.add(line.trim());
+                line = reader.readLine();
+            }
+        }
+
+        return iLeappPathsToProcess;
+    }
+
+    private void extractFilesFromImage(Content dataSource, List<String> iLeappPathsToProcess, Path moduleOutputPath) {
```
Comment (kellykelly3) [inline Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappAnalyzerIngestModule.java:384]: method header.

File: Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappAnalyzerIngestModule.java [hunk 5]
```diff
@@ -264,4 +357,109 @@ private void addILeappReportToReports(Path iLeappOutputDir, Case currentCase) {
...
+            for (AbstractFile iLeappFile : iLeappFiles) {
+                Path parentPath = Paths.get(moduleOutputPath.toString(), iLeappFile.getParentPath());
+                File fileParentPath = new File(parentPath.toString());
+
+                extractFileToOutput(dataSource, iLeappFile, fileParentPath, parentPath);
+            }
+        }
+    }
+
+    private void extractFileToOutput(Content dataSource, AbstractFile iLeappFile, File fileParentPath, Path parentPath) {
```
Comment (kellykelly3) [inline Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappAnalyzerIngestModule.java:420]: method header

File: Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappAnalyzerIngestModule.java [hunk 6]
```diff
@@ -264,4 +357,109 @@ private void addILeappReportToReports(Path iLeappOutputDir, Case currentCase) {
...
+                        try {
+                            Files.createDirectories(Paths.get(parentPath.toString(), iLeappFile.getName()));
+                        } catch (IOException ex) {
+                            logger.log(Level.INFO, String.format("Error creating iLeapp output directory %s", parentPath.toString()), ex);
+                        }
+                    }
+                }
+    }
+    
+    private void writeiLeappFile(Content dataSource, AbstractFile iLeappFile, String parentPath) {
```
Comment (kellykelly3) [inline Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappAnalyzerIngestModule.java:449]: method header.

**Review by kellykelly3 — CHANGES_REQUESTED** [review]

File: Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappAnalyzerIngestModule.java (hunk 1 above)
Comment (markmckinnon) [reply Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappAnalyzerIngestModule.java:229]: Done

File: Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappAnalyzerIngestModule.java (hunk 2 above)
Comment (markmckinnon) [reply Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappAnalyzerIngestModule.java:328]: Done

File: Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappAnalyzerIngestModule.java (hunk 3 above)
Comment (markmckinnon) [reply Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappAnalyzerIngestModule.java:392]: Done

File: Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappAnalyzerIngestModule.java (hunk 4 above)
Comment (markmckinnon) [reply Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappAnalyzerIngestModule.java:418]: Done

File: Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappAnalyzerIngestModule.java (hunk 5 above)
Comment (markmckinnon) [reply Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappAnalyzerIngestModule.java:461]: Done

File: Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappAnalyzerIngestModule.java (hunk 6 above)
Comment (markmckinnon) [reply Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappAnalyzerIngestModule.java:496]: Done

**Pull request merged**